### DISCUSSION TITLE
Will alevin-fry support custom library structure?

### DISCUSSION THREAD

**Discussion by diyang1354** [COMBINE-lab/alevin-fry#40] [Q&A] [answered]
Hi,
I'm wondering if alevin-fry can support customized single cell protocols? It would be great if we can change the barcode range and umi range accordingly. Thanks.
Best,
Di

**Comment by rob-p** [comment, marked as answer]
Hi @diyang1354,

  Sorry to have missed this in the discussion!  Yes; this is already possible. You can pass custom geometry for the UMI, barcode, and read for subsequent processing with alevin-fry. Strangely, the best documentation for this currently is in the release notes for salmon w[here](https://github.com/COMBINE-lab/salmon/releases/tag/v1.4.0) this feature was added here. We should add the documentation of this syntax to the documentation proper. Please let me know if you have any further questions about this.

Best,
Rob

**Reply by diyang1354** [reply]
Thank you so much Rob. 
I've found another issue in the doc related to the **strandness**, and I would be grateful if you'd like to discuss this.
In the doc [here](https://combine-lab.github.io/alevin-fry-tutorials/2021/improving-txome-specificity/), library type is IU for chromium V3. However, in the doc  [here](https://salmon.readthedocs.io/en/latest/alevin.html) the library type is ISR for chromium V2. Is there a specific reason to set different library types for these two technologies? It seems that the library structure is exactly the same except for different umi length according to libstruct posted [here](https://teichlab.github.io/scg_lib_structs/methods_html/10xChromium3.html)
The strandness issue really confused me a lot when I try to use alevin with my library. 
My library basically looks like this:
![image](https://user-images.githubusercontent.com/51859749/158816857-c27b0e96-989a-47bd-b30f-dfbb92c58d38.png)
I have 150 bp paired-end data. The read 1 comes from the forward strand covering the cDNA. The read 2 comes from the reverse strand and the first 8bp is cell barcode followed by 8 bp umi. 
**Would my library be ISF then?** 
Best,
Di

**Reply by rob-p** [reply]
Hi @diyang1354,

  Thank you for bringing this up.  I can see how this can be confusing.  Let me give a short answer and long answer:
  
*  Short answer : Either library format (`IU` or `ISR`) in `salmon` is OK for both 10xV2 and 10xV3.

* Longer answer : Unlike the original `alevin` — the library type passed to `salmon` when generating a RAD file for `alevin-fry` _does not filter out_ any mappings.  Rather, in `alevin-fry`, any filtering mappings based on strandedness is performed in the `generate-permit-list` step, via the `-d, --expected-ori` options.  That is, if you pass `IU` or `ISR` during creation of the RAD file, you should get essentially the same output.  Whether or not you wish to filter out or keep "biological" reads that map in the reverse complement orientation is determined by whether you pass `fw` or `both` (respectively) to the `alevin-fry` `generate-permit-list` command.  We'll try to clarify this further in the documentation.

Best,
Rob
